runtime/src/bank_forks.rs

```diff
@@ -542,7 +542,7 @@ impl BankForks {
 
     pub fn prune_program_cache(&self, root: Slot) {
         if let Some(root_bank) = self.banks.get(&root) {
-            root_bank.prune_program_cache(root, root_bank.epoch());
+            root_bank.prune_program_cache_locked(root, root_bank.epoch(), self);
```

Comment:
is there any case where we could possibly come through here and root bank FG doesn't match the bank_forks we call in with? I don't really understand all the usages here well enough

Reply:
On snapshot load the root bank is initially deserialized with an empty program cache, however when initializing `BankForks::new_rw_arc()`, we fill in the forks graph as the `BankForks`.

Then `bank::_new_from_parent` will use the `transaction_processor` and thus the `program_cache / FG` from the parent bank so I believe this should be consistent. Also I don't think we should ever have multiple `BankForks` instances outside of tests.